src/generate/convert/control_flow.rs

```diff
@@ -9,7 +9,7 @@ pub fn convert_cntrl_flow(ast: &ASTTy, imp: &mut Imports, state: &State, ctx: &C
     Ok(match &ast.node {
         NodeTy::IfElse { cond, then, el } => match el {
             Some(el) => Core::IfElse {
-                cond: Box::from(convert_node(cond, imp, state, ctx)?),
+                cond: Box::from(convert_node(cond, imp, &state.last_ret(false), ctx)?),
```

Comment:
Feels a bit finicky to add logic here for the if-else construct (though it is the only construct where the `last_ret` is propagated, see `convert_node` fun).

Reply:
I guess the `if-else` is the only branching language construct, so it makes sense that this is the only place that this happens. Still a bit weird to have it located here but I can't think of another solution.